test/elixir/test/mango/24_text_paginated_test.exs

```diff
@@ -0,0 +1,127 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+defmodule PaginatedResultsTest do
+  alias Couch.Test.Utils
+  use CouchTestCase
+
+  @db_name "paginated-results"
+  # Great enough to make faster systems busy while running the
+  # query.
+  @num_docs 9_999
+  @updates 25
+
+  setup do
+    MangoDatabase.recreate(@db_name)
+    fields = [
+      %{"name" => "_id", "type" => "string"},
+      %{"name" => "name", "type" => "string"}
+    ]
+    MangoDatabase.create_text_index(
+      @db_name,
+      analyzer: "keyword",
+      default_field: %{},
+      selector: %{},
+      fields: fields,
+      index_array_lengths: true
+    )
+
+    docs =
+      0..@num_docs
+      |> Enum.map(fn doc_id ->
+        %{
+          "_id" => hex8(doc_id),
+          "name" => Utils.random_name("db")
+        }
+      end)
+
+    MangoDatabase.save_docs(@db_name, docs)
+    :ok
+  end
+
+  defp hex8(i) do
+    :io_lib.format("~8.16.0B", [i]) |> IO.iodata_to_binary()
+  end
+
+  defp do_query(delay, selector, opts) do
+    Task.async(fn ->
+      Process.sleep(delay)
+      MangoDatabase.find(@db_name, selector, opts)
+    end)
+  end
+
+  defp do_updates(pause, doc_id) do
+    Task.async(fn ->
+      for i <- 0..@updates do
+        doc = MangoDatabase.open_doc(@db_name, doc_id)
+        updated_doc = %{
+          "_id" => doc_id,
+          "_rev" => doc["_rev"],
+          "update" => i,
+          "name" => "foobar"
+        }
+
+        MangoDatabase.save_doc(@db_name, updated_doc)
+        Process.sleep(pause)
+      end
+    end)
+  end
+
+  test "query with lot of results" do
+    selector = %{"_id" => %{"$lte": hex8(1000)}}
+    # 200 is the maximum for `text` searches.
+    {:ok, docs} = MangoDatabase.find(@db_name, selector, limit: 200)
+    assert length(docs) == 200
+  end
+
+  test "no duplicates on interleaved updates" do
+    # Give ~500 ms head start for the updates before running the
+    # query.
+    query = do_query(500, %{"name" => "foobar"})
```

Comment:
i get error here. i'm running `make elixir-search tests=test/elixir/test/mango/24_text_paginated_test.exs`
is there anything additional i need to do?
```
    error: undefined function do_query/2 (expected PaginatedResultsTest to define such a function or for it to be imported, but none are available)
    │
 89 │     query = do_query(500, %{"name" => "foobar"})
    │             ^^^^^^^^
    │
    └─ test/elixir/test/mango/24_text_paginated_test.exs:89:13: PaginatedResultsTest."test no duplicates on interleaved updates"/1
```

Reply:
Interesting